src/mlpack/methods/hmm/hmm_train_main.cpp

```diff
@@ -62,8 +63,8 @@ PROGRAM_INFO("Hidden Markov Model (HMM) Training",
         "@doxygen/classmlpack_1_1hmm_1_1HMM.html"));
 
 PARAM_STRING_IN_REQ("input_file", "File containing input observations.", "i");
-PARAM_STRING_IN("type", "Type of HMM: discrete | gaussian | gmm.", "t",
-    "gaussian");
+PARAM_STRING_IN("type", "Type of HMM: discrete | gaussian | gmm | gmm_diag.",
+    "t", "gaussian");
```

Comment:
Maybe we should use the name `diag_gmm` instead?  Also, I'd suggest putting `diag_gmm` before `gmm`, since that means they are labeled in order of ascending complexity/training time.